topology/mask_impl.h

```diff
@@ -87,40 +87,318 @@ Mask< D >::outside( const Box< D >& b ) const
   return false;
 }
 
-template < int D >
+template <>
 bool
-BoxMask< D >::inside( const Position< D >& p ) const
+BoxMask< 2 >::inside( const Position< 2 >& p ) const
 {
-  return ( p >= lower_left_ ) && ( p <= upper_right_ );
+  // If the box is not rotated we just check if the point is inside the box.
+  if ( not is_rotated_ )
+  {
+    return ( lower_left_ <= p ) && ( p <= upper_right_ );
+  }
+
+  // If we have a rotated box, we rotate the point down to the unrotated box,
+  // and check if it is inside said unrotated box.
+
+  // The new x, y values are calculated using a rotation matrix:
+  // [new_x, new_y] = R(-azimuth)*[x - x_c, y - y_c]
+  // where R(-t) = [cos(t) sin(t); -sin(t) cos(t)]
+
+  // See https://en.wikipedia.org/wiki/Rotation_matrix for more.
+
+  const double cntr_x = ( upper_right_[ 0 ] + lower_left_[ 0 ] ) * 0.5;
```

Comment:
If I am not entirely off, we need to check quite often if a point is inside a (fixed) box. Wouldn't it make more sense to precompute all values that do not depend on `p` in the box constructor?